working_villagers/jobs/guard.lua

```diff
@@ -3,12 +3,29 @@ local co_command = working_villages.require("job_coroutines").commands
 
 --modes: stationary,escort,patrol,wandering
 
+local guard_tools = {
+	["default:sword_mese"] = 1,
+	["default:sword_wood"] = 1,
+	["default:sword_steel"] = 1,
+	["default:sword_stone"] = 1,
+	["default:sword_bronze"] = 1,
+	["default:sword_diamond"] = 1,
+}
 working_villages.register_job("working_villages:job_guard", {
 	description      = "guard (working_villages)",
 	long_description = "I'm here on guard defending against all those whom I see as enemies.",
 	inventory_image  = "default_paper.png^memorandum_letters.png", --TODO: sword/bow/shield
+	-- TODO guard should wield a sword (and shield if appropriate mod is loaded)
 	jobfunc = function(self)
-		local guard_mode = self:get_job_data("mode") or "stationary"
+		local stack  = self:get_wield_item_stack()
+		if stack:is_empty() then
+		self:move_main_to_wield(function(name)
+  			return guard_tools[name] ~= nil
+		end)
+		end
+		-- how to change the mode ?
```

Comment:
Guards could be wandering, on patrol or stationary, depending on their surroundings and other villagers.